commands/show.go

```diff
@@ -33,49 +37,75 @@ func runShowBug(cmd *cobra.Command, args []string) error {
 
 	firstComment := snapshot.Comments[0]
 
-	// Header
-	fmt.Printf("[%s] %s %s\n\n",
-		colors.Yellow(snapshot.Status),
-		colors.Cyan(snapshot.HumanId()),
-		snapshot.Title,
-	)
-
-	fmt.Printf("%s opened this issue %s\n\n",
-		colors.Magenta(firstComment.Author.DisplayName()),
-		firstComment.FormatTimeRel(),
-	)
-
-	var labels = make([]string, len(snapshot.Labels))
-	for i := range snapshot.Labels {
-		labels[i] = string(snapshot.Labels[i])
-	}
-
-	fmt.Printf("labels: %s\n\n",
-		strings.Join(labels, ", "),
-	)
-
-	// Comments
-	indent := "  "
+	if showFieldsQuery==nil {
+		// Header
+		fmt.Printf("[%s] %s %s\n\n",
+			colors.Yellow(snapshot.Status),
+			colors.Cyan(snapshot.HumanId()),
+			snapshot.Title,
+		)
 
-	for i, comment := range snapshot.Comments {
-		var message string
-		fmt.Printf("%s#%d %s <%s>\n\n",
-			indent,
-			i,
-			comment.Author.DisplayName(),
-			comment.Author.Email,
+		fmt.Printf("%s opened this issue %s\n\n",
+			colors.Magenta(firstComment.Author.DisplayName()),
+			firstComment.FormatTimeRel(),
 		)
 
-		if comment.Message == "" {
-			message = colors.GreyBold("No description provided.")
-		} else {
-			message = comment.Message
+		var labels = make([]string, len(snapshot.Labels))
+		for i := range snapshot.Labels {
+			labels[i] = string(snapshot.Labels[i])
 		}
 
-		fmt.Printf("%s%s\n\n\n",
-			indent,
-			message,
+		fmt.Printf("labels: %s\n\n",
+			strings.Join(labels, ", "),
 		)
+
+		// Comments
+		indent := "  "
+
+		for i, comment := range snapshot.Comments {
+			var message string
+			fmt.Printf("%s#%d %s <%s>\n\n",
+				indent,
+				i,
+				comment.Author.DisplayName(),
+				comment.Author.Email,
+			)
+
+			if comment.Message == "" {
+				message = colors.GreyBold("No description provided.")
+			} else {
+				message = comment.Message
+			}
+
+			fmt.Printf("%s%s\n\n\n",
+				indent,
+				message,
+			)
+		}
+	} else {
+		unknownFields:=""
+		err:=false
```

Comment:
This very not like go. In go, an `err` variable is very much expected to be of type `error`, not a boolean. I think you can simply return an error as soon as you find a malformed query (the `default` case).

Reply:
Made it shorter, then no more need of this ambiguous err flag.